git_wrapper/base.py

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env python
+'''This module acts as an interface for common git tasks'''
+
+import git
+
+
+class GitWrapperBase(object):
+    '''Provides a wrapper to interact with a git repository'''
+
+    def __init__(self, path='', repo=None):
+        """Constructor for GitUtilBase object
+            :param str path: Path to a git repo Default('')
+            :param git.Repo repo: An already construnted git.Repo object to use Default(None)
+        """
+        self.__repo = None  # Added to clear pylint warnings
+        if repo is not None:
+            self.repo = repo
+        elif path != '':
+            self.repo = git.Repo(path)
+        else:
+            raise Exception('No path or repo given')
+
+    @property
+    def repo(self):
+        """Returns the git repo for a given path
+            :return A reference to the internal git.Repo object
+        """
+        return self.__repo
+
+    @repo.setter
+    def repo(self, new_repo):
+        """Allows for use of an already constructed repo
+            :param git.Repo new_repo: An already construnted git.Repo object to use
+        """
+        self.__repo = new_repo
+
+    def get_git_command(self):
+        '''Returns the git command for a given repo'''
+        return self.__repo.git
+
+    def get_remote_names(self):
+        """Returns a list of remotes for a given repo
+            :return A list of utf-8 encoded remote names
+        """
+        return [x.name for x in self.__repo.remotes]
```

Comment:
If you are using properties, I think it would be more consistent to substitute all calls to self.__{property} with self.{property}